### ISSUE TITLE
Support UNIX sockets

### ISSUE THREAD

**Issue by david-macmahon** [JuliaDatabases/Redis.jl#84]
redis-server can use only UNIX sockets (i.e. no TCP sockets) by setting `port` to 0 and `unixsocket` to the pathname of a UNIX socket.  Redis.jl only supports TCP sockets currently so it cannot access such a server.  Not a big problem if one can configure the server to also bind to a TCP port on `localhost`, but a very big problem if one does not control the redis-server configuration.

**Comment by jkaye2012** [comment]
Are you familiar with how the Julia networking libraries work in the current version of the standard lib? I took a quick look at it seems a bit odd that `connect` has an overload that returns `PipeEndpoint` instead of `TCPSocket`, but the documentation on what one can actually do with that endpoint is severely lacking.

**Comment by david-macmahon** [comment]
Hmm, I didn't realize Redis.jl did its own socket management. I had assumed (uh-oh!) that it was using a C client under the covers.

It looks like `Base.PipeEndpoint` and `Sockets.TCPSocket` are both subtypes of `Base.LibuvStream`.  Once can use `methodswith(Base.PipeEndpoint)` to see that it doesn't have any special methods.  `TCPSocket` has a few methods, but most (all?) of the interesting stuff shows up in `methodswith(Base.LibuvStream)`.

**Comment by jkaye2012** [comment]
Yep, this is a pure Julia implementation (for better or for worse 😉). In
that case we may just be able to change the types in connection.jl and
everything should work itself out (or blow up if somehow we are dependent
on tcp, but I doubt it).

**Comment by david-macmahon** [comment]
I think `RedisConnection` might need a new `RedisConnection(; unixsocket)` (or similar) method, but I think that would effectively be the same as the `RedisConnection(;host, port, ...)` method since the kwargs don't count towards dispatch.  Might have to roll the new keyword into the existing method and somehow discern what the caller wants.

**Comment by jkaye2012** [comment]
Yeah, you're right. I thought the two connect methods accepted the same
argument, but it makes more sense that they don't. Either way should not be
difficult to do.

**Comment by david-macmahon** [comment]
I guess one could use `RedisConnection(; host="localhost", port=6379, unixsocket="")` and then use `unixsocket` if it is not empty, otherwise use `host`/`port`.  Maybe just name it `socket` rather than `unixsocket` so it doesn't sound so strange for non-unix systems?  I don't know what the windows equivalent it called.
